cpp/src/datetime/datetime_ops.cu

```diff
@@ -155,6 +161,18 @@ struct is_leap_year_op {
   }
 };
 
+// Extract the number of days of the month
+struct days_in_month_op {
+  template <typename Timestamp>
+  CUDA_DEVICE_CALLABLE int16_t operator()(Timestamp const ts) const
+  {
+    using namespace cuda::std::chrono;
+    auto const days_since_epoch = floor<days>(ts);
+    auto const date             = year_month_day(days_since_epoch);
+    return static_cast<int16_t>(days_in_month(date.month(), date.year().is_leap()));
```

Comment:
Not sure if more efficient, but would using [`year_month_day_last`](https://en.cppreference.com/w/cpp/chrono/year_month_day_last)
 be any more concise?

Reply:
Good point - this is how I created the reference answer in tests, let me benchmark which method is more efficient. I would opt for `ymdl` if they are roughly on the same order of efficiency.